plonky3/src/circuit_builder.rs

```diff
@@ -225,13 +172,23 @@ impl<'a, T: FieldElement> PowdrCircuit<'a, T> {
                     }
                 }
             }
-            AlgebraicExpression::PublicReference(_) => unimplemented!(
-                "public references are not supported inside algebraic expressions in plonky3"
-            ),
+            AlgebraicExpression::PublicReference(id) => {
+                assert!(
```

Comment:
This could be more concise by using `Iter::find`

Reply:
Clippy recommended to use `any` is that equivalent?

Reply:
Here you assert any, then find the position, then return the element at that position. The three can be done with a single call to find.